### PR TITLE
Make createdAt a universal field

### PR DESCRIPTION
**Pull request by oetherington**
This PR makes `createdAt` a universal field which exists on all current and future collections. This should allow us to do the Postgres migration with 0 downtime.

On the dev database, the migration takes ~20 mins, virtually all of which is the `Votes` and `Revisions` collections. This will need to be run before and after deploying (and should be trivially fast after).

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1202952823355469) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/server/migrations/2022-09-08-giveAllCollectionsCreatedAt.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    unmigratedDocumentQuery: {
+      createdAt: {$exists: false},
+    },
+    migrate: async (documents: T[]) => {
+      const updates = documents.map((document: T) => ({
+        updateOne: {
+          filter: {_id: document._id},
+          update: {
+            $set: {
+              createdAt: new Date(document[existingField] ?? Date.now()),
```
Comment (jpaddison3) [inline packages/lesswrong/server/migrations/2022-09-08-giveAllCollectionsCreatedAt.ts:31]: I feel like this should be 1970-01-01 as a better null date.

**Review by jpaddison3 — APPROVED** [review]
LGTM

File: packages/lesswrong/lib/collectionUtils.ts [hunk 2]
```diff
@@ -198,9 +198,14 @@ export function schemaDefaultValue<T extends DbObject>(defaultValue: any): Parti
...
 
-export function addUniversalFields<T extends DbObject>({ collection, schemaVersion=1 }: {
+export function addUniversalFields<T extends DbObject>({
+  collection,
+  schemaVersion = 1,
+  createdAtOptions = {},
+}: {
   collection: CollectionBase<T>,
   schemaVersion?: number
+  createdAtOptions?: Record<string, any>,
```
Comment (b0b3rt) [inline packages/lesswrong/lib/collectionUtils.ts:208]: Can we type this to at least be limited to only those fields (and their respective types) we're currently using it for, if not the full list of permissible options?

Comment (b0b3rt) [reply packages/lesswrong/lib/collectionUtils.ts:208]: Looks like `editableBy`, `insertableBy`, `viewableBy`, and `canRead`.

**Comment by oetherington** [comment]
@b0b3rt I've just run the migration on the EA Forum. In theory, you don't actually need to run it on LW until we switch to postgres, but it would probably be good to do it sooner anyway. For reference, it took about 2.5 hours for us.

**Comment by Raemon** [comment]
Note this'll probably take significantly longer for LW based on previous migrations (last time I think it was 3x longer on LW, maybe 4x. Sounds like it might be like 7 hours?)

**Comment by jpaddison3** [comment]
Note that Ollie was running on his local machine in the UK, whereas you will be running closer to us-east-1. You could also be smarter and run against an aws server running in us-east-1 itself and get significantly faster performance.

**Pull request merged**